File: 08/rb/08.rb
#!/usr/bin/env ruby

layers = *File.read('input').chomp.chars.each_slice(25*6)

# part 1
p layers.map{|x|[x.count(?0), x.count(?1)*x.count(?2)]}.min[1]

# part 2
puts Array.new(25) {|x|
    Array.new(6) {|y|
        layers.map{|z| z[y*25+x]}.join.delete(?2)[0].tr '01', ' #'
    }
}.transpose.map &:join

File: 08/hs/08.hs
#!/usr/bin/env runhaskell

import Control.Monad
import Data.Function
import Data.List
import Data.List.Split

part1 = liftM2 (*) (count '1') (count '2') . minimumBy (compare `on` count '0') . map concat
    where count x = length . filter (==x)

part2 = foldl1 $ zipWith.zipWith $ f
    where f '2' x = g x
          f x   _ = g x
          g '0' = ' '
          g '1' = '#'
          g x   = x

main = do
    layers <- chunksOf 6 . chunksOf 25 . init <$> readFile "input"
    putStrLn . show    $ part1 layers
    putStrLn . unlines $ part2 layers
